daemon/api.go

```diff
@@ -2637,3 +2639,80 @@ func getLogs(c *Command, r *http.Request, user *auth.UserState) Response {
 		follow:     follow,
 	}
 }
+
+type appInstruction struct {
+	Action string   `json:"action"`
+	Names  []string `json:"names"`
+	client.StartOptions
+	client.StopOptions
+	client.RestartOptions
+}
+
+func postApps(c *Command, r *http.Request, user *auth.UserState) Response {
+	var inst appInstruction
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&inst); err != nil {
+		return BadRequest("cannot decode request body into service operation: %v", err)
+	}
+	if len(inst.Names) == 0 {
+		// on POST, don't allow empty to mean all
+		return BadRequest("cannot perform operation on services without a list of services to operate on")
+	}
+
+	st := c.d.overlord.State()
+	appInfos, rsp := appInfosFor(st, inst.Names, appInfoOptions{service: true})
+	if rsp != nil {
+		return rsp
+	}
+	if len(appInfos) == 0 {
+		// can't happen: appInfosFor with a non-empty list of services
+		// shouldn't ever return an empty appInfos with no error response
+		return InternalError("no services found")
+	}
+
+	argv := make([]string, 2, 3+len(appInfos))
+	argv[0] = "systemctl"
+	argv[1] = inst.Action
+
+	switch inst.Action {
+	case "start":
+		if inst.Enable {
+			argv[1] = "enable"
+			argv = append(argv, "--now")
+		}
+	case "stop":
+		if inst.Disable {
+			argv[1] = "disable"
+			argv = append(argv, "--now")
+		}
+	case "restart":
+		if inst.Reload {
+			argv[1] = "try-reload-or-restart"
+		}
+	default:
+		return BadRequest("unknown action %q", inst.Action)
+	}
+
+	snapNames := make(map[string]bool, len(appInfos))
+	names := make([]string, len(appInfos))
+	for i, svc := range appInfos {
+		argv = append(argv, svc.ServiceName())
+		snapName := svc.Snap.Name()
+		names[i] = snapName + "." + svc.Name
+		snapNames[snapName] = true
+	}
+
+	desc := fmt.Sprintf("%s of %v", inst.Action, names)
+
+	st.Lock()
+	defer st.Unlock()
+	if err := snapstate.CheckChangeConflictMulti(st, snapNames, nil); err != nil {
+		return Conflict(err.Error())
```

Comment:
what's the resource? and what's state?   the conflict here is transient while my reading of 409 is that it would be about a permanent state of the resource,  so usually asking again wouldn't work either (unless there's some approach that is happy if the state is the same by different paths but that's not relevant here)